app/assets/javascripts/discourse/app/routes/discourse.js

```diff
@@ -74,6 +74,14 @@ const DiscourseRoute = Route.extend({
     }
   },
 
+  isAnotherUsersPage(user) {
+    if (!this.currentUser) {
+      return true;
+    }
+
+    return user.username !== this.currentUser.username;
+  },
```

Comment:
Personally, I would extract this function into the `User` class. Something like `User.isCurrentUser(user)`

Reply:
I'll be happy to change it to another function if there are pros in another implementation, or if there are some problems with my implementation.

By now, I prefer my version. Here is why.

A user may look at his own page or at a page of another user. Very often, we need to know if a user _**looks at a page of another user**_. Take a look how the name of my function is explicitly saying just _exactly_ what we're checking. Explicit descriptive names are very important.

Moreover, we check not only if this is the current user, we actually need to check two things:
- if this is an anonymous user (for an anonymous user, any page is a page of another user)
- and only if it's not an anonymous user, we check if it is his page

I would even implement this function like this, more explicitly, I would extract a variable to make my intentions obvious:
```javascript
  isAnotherUsersPage(user) {
    const isAnonimousUser = !this.currentUser;
    if (isAnonimousUser) {
      return true;
    }

    return user.username !== this.currentUser.username;
  },
```

Only this part of the function can go to `User.isCurrentUser(user)`, not 100% sure if we need to extract it, but maybe yes:
```javascript
  isCurrentUser(user) {
    return user.username !== this.currentUser.username;
  }
```

By the way, if we extract that part, my function will look better
```javascript
  isAnotherUsersPage(user) {
    const isAnonimousUser = !this.currentUser;

    return isAnonimousUser || User.isCurrentUser(user)
  },
```
@tgxworld What do you think? Also, maybe there are some problems with my function that I don't see?

Reply:
Thanks for the detailed explanation again :) The last example looks good to me actually. Actually I wonder if we already have such a function lying around somewhere.

Reply:
> Actually I wonder if we already have such a function lying around somewhere.
I feel now like I saw something like this, going to look for it.

Reply:
> I feel now like I saw something like this, going to look for it.

I had a similar use case today but couldn't find it. Anyway, please ignore my comments here because I realised that the `currentUser` object is not injected into models automatically.